### ISSUE TITLE
Pretrained Models

### ISSUE THREAD

**Issue by Hepta-Col** [facebookresearch/pytorchvideo#200]
![image](https://user-images.githubusercontent.com/62273225/172222701-653c2bdf-d7ec-4c31-9e1b-b33460a4221b.png)
I want to use a pretrained I3D to generate video features. What should I do after I download the "link"? How do I deal with the downloaded PYTH file? If it is the checkpoint of the model, then for those models that are not included in PyTorch Hub, like I3D, C2D, how should I load the checkpoint onto them?

**Comment by Hepta-Col** [comment]
in addition, what should I do if I just want an intermediate representation of the video rather than the final logits?

**Comment by lyttonhao** [comment]
Hi, I3D and C2D models are supported by our PyTorch Hub, as shown in this [file](https://github.com/facebookresearch/pytorchvideo/blob/main/pytorchvideo/models/hub/resnet.py).

For feature extraction (similar to https://github.com/facebookresearch/pytorchvideo/issues/188), you could make `head=None for kwargs to remove the prediction heads when [creating models](https://github.com/facebookresearch/pytorchvideo/blob/e3fc8fb4d1ea8dfb0aea3b1064b58fa84c97b97c/pytorchvideo/models/resnet.py#L646)

**Comment by yyccli** [comment]
Hi, i'm doing this as you suggested, but when i passed `head=None` using `torch.hub` to build models, i encounted such problems:

![image](https://user-images.githubusercontent.com/25741435/192726205-32fe64bc-aa4d-4831-b0ff-b1eb3976f66c.png)

![image](https://user-images.githubusercontent.com/25741435/192726272-61f2eca1-563f-46ee-83fa-fec5098c3c95.png)

It seems that the model i want to build actually does't have a `head` module, but the pre-trained model still tries to load weights into it.

I'm not sure if i do something wrong?

**Comment by Camillo4eyes** [comment]
@yyccli i have the same error, you solved?

**Comment by daviduarte** [comment]
Fast workaround: 
Edit the file **pytorchvideo/models/hub/resnet.py**

In the function 

`def _resnet(
    pretrained: bool = False,
    progress: bool = True,
    checkpoint_path: str = "",
    model_builder: Callable = create_resnet,
    **kwargs: Any,
) -> nn.Module:
    model = model_builder(**kwargs)
    if pretrained:
        # All models are loaded onto CPU by default
        checkpoint = load_state_dict_from_url(
            checkpoint_path, progress=progress, map_location="cpu"
        )
        state_dict = checkpoint["model_state"]
        model.load_state_dict(state_dict)
    return model`

drop the weights you are not using (those appearing in the error). In my case, I'm dropping **blocks.5.proj.weight** and **blocks.5.proj.bias**. So, the function will be: 

`def _resnet(
    pretrained: bool = False,
    progress: bool = True,
    checkpoint_path: str = "",
    model_builder: Callable = create_resnet,
    **kwargs: Any,
) -> nn.Module:
    model = model_builder(**kwargs)
    if pretrained:
        # All models are loaded onto CPU by default
        checkpoint = load_state_dict_from_url(
            checkpoint_path, progress=progress, map_location="cpu"
        )
        state_dict = checkpoint["model_state"]
        print(state_dict['blocks.5.proj.weight'])
        del state_dict['blocks.5.proj.weight']
        del state_dict['blocks.5.proj.bias']
        #exit()
        model.load_state_dict(state_dict)
    return model`

Corrected sentence: If you are extracting the features at a different height, you have to manually change the code to reflect your change.
